sdk/nodejs/appservice/zMixins_http.ts

```diff
@@ -104,11 +135,18 @@ export class HttpEventSubscription extends mod.EventSubscription<mod.Context<Htt
             ...args,
             location,
             resourceGroupName,
+            hostSettings: {
+                ...args.hostSettings,
+                extensions: { http: args.extensionSettings },

```

Comment:
i'm a little confused by this. why can't hte caller just provide this 'extensionSetting' in the 'hostSettings object tehy can provide (since we just pass that along)

Reply:
I started with that, but then I thought that passing Blob options doesn't make sense for HTTP trigger, so why expose the full surface of options when only a small subset is meaningful in this context.

Reply:
we don't have to expose the full set.  But we can say, for http events, that the args has a `hostSettings` object with the appropriate field.  One field would be `extensions` which has a `http` arg. 

Basically, from the TS perspective, we only expose what we think is relevant, but we keep the shape in line with the actual shape that is passed down.

Reply:
`hostSettings` is defined in `CallbackFunctionAppArgs`. Does it mean that `hostSettings` should become generic with a different type of `extension` in different event subscriptions?

Reply:
It doesn't even need to be generic (though that's one way to do this).  TS is *very* lenient here.  So just do this:

```ts
interface HttpHostSettings extends HostSettings {
   // add your new fields.
}

//...

export type HttpEventSubscriptionArgs = ... {
    hostSettings?: HttpHostSettings;
}>;
```

Basically, you're saying that for an http event subscription, the hostSettings property has these additional optional props.  This is extremely common in TypeScript and would be very much prefered here for me.

Does that make sense? :)  Sorry for the major delays in getting back to you.  I'm traveling and am on vacation :)

Reply:
Great! I still need to learn the TypeScript way of doing things. Thanks for explaining. Have a look at the latest version.
Enjoy your vacation :)